server/app/services/geo/AddressLocation.java

```diff
@@ -0,0 +1,42 @@
+package services.geo;
+
+import com.google.auto.value.AutoValue;
+
+@AutoValue
+public abstract class AddressLocation {
+  public static Builder builder() {
+    return new AutoValue_AddressLocation.Builder();
+  }
+  // TODO: consistent nameing long/lat or x/y?
+  /**
+   * this method returns a coordinate on the x axis of the coordinate system (longitude)
+   *
+   * @return Long
+   */
+  public abstract Long getX();
```

Comment:
suggestion (non-blocking): my vote would be to call this longitude and call "Y" latitude